### ISSUE TITLE
Typo in Amazon Linux 2023 `nodeadm` examples

### ISSUE THREAD

**Issue by rblaine95** [terraform-aws-modules/terraform-aws-eks#2985]
There's a typo in the Amazon Linux 2023 `nodeadm` examples that caused me a significant amount of confusion.
`v1alpha` is meant to be `v1alpha1`

---

## Original Description

Trying to create AL2023 EKS Managed Nodes.
Want to use the `nodeadm` to enable graceful shutdown as well as some node labels for workload scheduling.
The EC2 instance starts up, but then cloud-init prints a warning saying `unhandled unknown content-type (application/node.eks.aws` - very similar to #2954 and the node never joins the cluster.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.
May be similar to #2954

## ⚠️ Note

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

- Module version [Required]: `20.8.4`
- Terraform/Tofu and Provider/s version:
```
$ tofu providers --version
OpenTofu v1.6.2
on darwin_arm64
+ provider registry.opentofu.org/cyrilgdn/postgresql v1.22.0
+ provider registry.opentofu.org/datadog/datadog v3.37.0
+ provider registry.opentofu.org/hashicorp/aws v5.40.0
+ provider registry.opentofu.org/hashicorp/cloudinit v2.3.3
+ provider registry.opentofu.org/hashicorp/external v2.3.3
+ provider registry.opentofu.org/hashicorp/helm v2.12.1
+ provider registry.opentofu.org/hashicorp/http v3.4.2
+ provider registry.opentofu.org/hashicorp/kubernetes v2.27.0
+ provider registry.opentofu.org/hashicorp/local v2.4.1
+ provider registry.opentofu.org/hashicorp/null v3.2.2
+ provider registry.opentofu.org/hashicorp/random v3.6.0
+ provider registry.opentofu.org/hashicorp/time v0.10.0
+ provider registry.opentofu.org/hashicorp/tls v4.0.5
+ provider registry.opentofu.org/integrations/github v5.45.0
+ provider registry.opentofu.org/mongodb/mongodbatlas v1.15.1
+ provider registry.opentofu.org/tailscale/tailscale v0.15.0
```

## Reproduction Code [Required]

```terraform
eks_managed_node_group_defaults = {
  ami_type                       = "AL2023_x86_64_STANDARD"
  platform                       = "al2023"
  use_latest_ami_release_version = false

  metadata_options = {
    http_endpoint               = "enabled"
    http_tokens                 = "required"
    http_put_response_hop_limit = 2
  }

  update_config = {
    max_unavailable = 1
  }
}

eks_managed_node_groups = {
  spot = {
    name          = "eks-cluster-spot"
    ami_id        = "ami-04ba0d9025353715f"
    capacity_type = "SPOT"
    subnet_ids    = var.eks_subnet_ids
    min_size      = 0
    max_size      = 10
    desired_size  = 1

    instance_types = [
      "t3a.xlarge",
      "t3.xlarge",
      "t2.xlarge",
      "m3.xlarge",
      "m4.xlarge",
      "m5.xlarge",
      "m5a.xlarge"
    ]

    enable_bootstrap_user_data = true
    cloudinit_pre_nodeadm = [{
      content_type = "application/node.eks.aws"
      content           = <<-EOT
        ---
        apiVersion: node.eks.aws/v1alpha
        kind: NodeConfig
        spec:
          kubelet:
            config:
              shutdownGracePeriod: 5m
              shutdownGracePeriodCriticalPods: 2m
            flags:
              - "--node-labels=node.kubernetes.io/lifecycle=spot,role=spot"
      EOT
    }]
  }
}
```

## Expected behavior

Node should start up and join the EKS cluster with the desired Node Config.

## Actual behavior

Node never joins the EKS cluster

**Comment by rblaine95** [comment]
Maybe I'm doing something stupid?

I'm basingmy `cloudinit_pre_nodeadm` config on what is in the [examples](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v20.8.4/examples/eks_managed_node_group/main.tf#L97-L112)

**Comment by rblaine95** [comment]
Alright, looks like that warning is a Red Herring 🥳 

Something is definitely up with _my_ `cloudinit_pre_nodeadm` as, removing it, the node successfully starts up and joins the cluster.

**Comment by bryantbiggs** [comment]
also, I think that node label is already automatically applied when you use `SPOT` with MNG, so you may not need that

**Comment by rblaine95** [comment]
Thank you @bryantbiggs 

A similar node label is added `eks.amazonaws.com/capacityType=SPOT`, but we use the one I'm adding in the above config.

Thank you for that URL, I think I see where I've gone wrong with the config I'm providing.

I see the example I linked is quite different to the AWS example you linked:
```yaml
# https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v20.8.4/examples/eks_managed_node_group/main.tf#L97-L112
---
apiVersion: node.eks.aws/v1alpha
kind: NodeConfig
spec:
  kubelet:
    config:
      shutdownGracePeriod: 30s
      featureGates:
        DisableKubeletCloudCredentialProviders: true
---
# https://github.com/awslabs/amazon-eks-ami/blob/c674960e68543ae9f257532b0c811986360b116f/nodeadm/example/node.eks.aws_v1alpha1_nodeconfig.yaml#L12-L23
apiVersion: node.eks.aws.v1alpha1
[...]
  kubelet:
    config:
      kind: KubeletConfiguration
      apiVersion: kubelet.config.k8s.io/v1beta1
      maxPods: 42
      clusterDNS:
        - 0.0.0.0
        - 1.1.1.1
    flags:
      - --v=2
      - --register-with-taints=foo=bar:NoSchedule
      - --node-labels=foo=bar
```
Will test and let you know shortly.

Thank you!

**Comment by rblaine95** [comment]
Yes, looks like it was exactly that!

This config works:
```yaml
---
apiVersion: node.eks.aws/v1alpha1
kind: NodeConfig
spec:
  kubelet:
    config:
      kind: KubeletConfiguration
      apiVersion: kubelet.config.k8s.io/v1beta1
      shutdownGracePeriod: 5m
      shutdownGracePeriodCriticalPods: 2m
    flags:
      - --node-labels=node.kubernetes.io/lifecycle=spot,role=spot
```

@bryantbiggs would you like me to open a PR fixing the [example](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v20.8.4/examples/eks_managed_node_group/main.tf#L97-L112) and maybe fleshing it out a little bit?

**Comment by bryantbiggs** [comment]
I believe the example is correct, and its pulled directly from the AMI repo https://awslabs.github.io/amazon-eks-ami/nodeadm/doc/examples/

**Comment by rblaine95** [comment]
Ah, I think I see the typo:
```diff
- apiVersion: node.eks.aws/v1alpha
+ apiVersion: node.eks.aws/v1alpha1
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 20.8.5 :tada:

**Comment by kappa8219** [comment]
In 20.8.4, I guess. But anyway 👍🏼
